crates/runtime/src/externref.rs

```diff
@@ -804,11 +805,20 @@ impl StackMapRegistry {
             min = std::cmp::min(min, range.start);
             max = std::cmp::max(max, range.end);
 
+            // Add a marker between functions indicating that this function's pc
+            // starts with no stack map so when our binary search later on finds
+            // a pc between the start of the function and the function's first
+            // stack map it doesn't think the previous stack map is our stack
+            // map.
+            if pc_to_stack_map.len() > 0 {
+                pc_to_stack_map.push((range.start, None));
+            }
+
             for info in infos {
                 assert!((info.code_offset as usize) < len);
                 pc_to_stack_map.push((
                     range.start + (info.code_offset as usize),
-                    Rc::new(info.stack_map.clone()),
+                    Some(Rc::new(info.stack_map.clone())),
```

Comment:
If `info.code_offset` is `0` then we could have duplicate entries (one `None` for the start of the function and one `Some` for this info).

I don't *think* this should ever happen though, so perhaps we can just assert that `info.code_offset > 0`?

The alternative is checking ahead of time whether the first `info.code_offset` is `0` (because they should be sorted iirc) and only adding the `None` entry if the first `info.code_offset` is not `0`.

Reply:
Oh nice catch, I refactored a bit which I think will handle this correctly, mind double-checking?

I also went ahead and did a small tweak where consecutive `None` entries are now coalesced.